nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py

```diff
@@ -190,51 +514,96 @@ def __init__(self):
         rospy.loginfo("initialized joint ctrl")
         self.gpr_ctrl = ManipulationGripperControl()
         rospy.loginfo("initialized gripper ctrl")
-        self.crt_ctrl = ManipulationCartesianControl()
-        rospy.loginfo("initialized cartesian ctrl")
+
+        self.right_cartesian = ManipulationCartesianControl("right")
+        self.left_cartesian = ManipulationCartesianControl("left")
 
         self.right_joint_states = [0, 0, 0, 0, 0, 0, 0]
         self.left_joint_states = [0, 0, 0, 0, 0, 0, 0]
         self.linear_joint_state = [0]
         self.right_ee_pose = None
         self.left_ee_pose = None
-        self.update_joint_states()
-        rospy.loginfo("joint states received, initialization successful")
-
-    def update_joint_states(self):
-        joint_states = rospy.client.wait_for_message("/joint_states", JointState)
-        p = joint_states.position
-        self.right_joint_states = [p[14], p[13], p[10], p[11], p[15], p[16], p[17]]
-        self.left_joint_states = [p[4], p[3], p[0], p[1], p[5], p[6], p[7]]
-        self.linear_joint_state = [p[8]]
-
-    def update_ee_pose(self):
-        """
-        robot_urdf = URDF.from_xml_string("urdf_str")
-        self.update_joint_states()
-        # base -> right ee
-        kdl_kin = KDLKinematics(robot_urdf, "base_link", "right_ee_link")
-        self.right_ee_pose = kdl_kin.forward(self.linear_joint_state + self.right_joint_states)
-
-        # base -> left ee
-        kdl_kin = KDLKinematics(robot_urdf, "base_link", "left_ee_link")
-        self.left_ee_pose = kdl_kin.forward(self.linear_joint_state + self.left_joint_states)
-        """
-        raise NotImplementedError()
+        rospy.loginfo("Manipulation initialization successful")
 
     def home(self):
-        # home left arm in joint space and home right arm in cart or joint space
-        # use self.jnt_ctrl.home() for now
-        raise NotImplementedError()
+        # CAUTION: This function should only ever home the arms. Don't add homing of other things here
+        # right gripper is openend on bootup by kinova. not sure where, but not in init
+        if not self.gpr_ctrl.close_right():
+            return False
+        if not self.jnt_ctrl.cmd_left_arm(self.jnt_ctrl.left_arm_home_joint_values):
+            return False
+        home_pose = GeometryPose()
```

Comment:
What joint does "home_pose" refer to? Is it always the end effector link of the arm? What if we want to adjust each joint to a specific position so we avoid any "flaring" of the arms. this can work for now but might be an issue later if the arm flaring is more of an issue

Reply:
in this context (the manipulators context since this is the manipulators pkg) home refers to the homing of the arms. we home left arm in joint space, but home the right arm in cartesian. this is bc the orientation lock only works for cartesian (as ive been told). we shouldnt command each joint on the right arm bc itd flip the box